### PR TITLE
[DEV-213517] Implement request signature integration

### PR DESCRIPTION
**Pull request by andresguedes**
Adding code to show how to get a request signature from remote API

### CODE REVIEW

**Review by marcosrivereto — APPROVED** [review]

**Review by renatodelpupo — APPROVED** [review]

**Review by lukevance — CHANGES_REQUESTED** [review]
Andre, can you please add some details in the Readme.md file about this new feature and how to configure it? Something like this section in the webview example app: https://github.com/TrustlyInc/trustly-webview-example/blob/4adbfa6713848502616cac99d493b9ded293972b/react-web-app/README.md#server-side-features

Specifically, please note:
* to enable this feature, a user must setup this dependency and link to trustly-nestjs-example project
* Note where in the app this integration occurs (RetrofitClient.kt line 27)
* in the EstablishData.kt remove line 8 or note the reference of where this is generated dynamically
* Anything else the user might need to know to set this up and run it, I haven't been able to get very far because I'm stuck with a Java runtime error, which leads me to...

It appears that the project now requires a higher version of the Java runtime? Please correct me if I'm wrong, but I'm getting this error when I try to build the app: 

```Build file '/Users/lukevance/sandbox_dev/trustly-android-example/app/build.gradle' line: 2

An exception occurred applying plugin request [id: 'com.android.application']
> Failed to apply plugin 'com.android.internal.application'.
   > Could not create an instance of type com.android.build.gradle.internal.dsl.ApplicationExtensionImpl$AgpDecorated.
      > Could not create an instance of type com.android.build.gradle.internal.dsl.LintImpl$AgpDecorated.
         > Could not generate a decorated class for type LintImpl$AgpDecorated.
            > com/android/tools/lint/model/LintModelSeverity has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0

```

**Review by lukevance — CHANGES_REQUESTED** [review]
I'm not sure my project settings/Java/Gradle errors are related to the app itself. I think the main thing is the instructions for setup, it's important that a user can follow two paths:
1. simple quickstart: just run the project with no backend integration and no requestSignature
2. add security: later when they have solved this part, the user may want to revisit the project to understand how to setup requestSignatures

See the [AC here ](https://trustly.atlassian.net/browse/DEV-213519) that mentions "uncommenting" the logic, that may not be the right approach but the user should easily understand how to go from step 1 above to step 2.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <resources>
-    <string name="app_name">PayWithMyBankSDKDemoAndroid</string>
+    <string name="app_name">TrustlySDKDemoAndroid</string>
```
[the remark this answers is not in the document]
Comment (andresguedes) [reply app/src/main/res/values/strings.xml:2]: 👍🏽

File: app/src/main/java/net/trustly/trustlysdkdemoandroid/ui/MainActivity.kt [hunk 2]
```diff
@@ -30,6 +44,35 @@ class MainActivity : AppCompatActivity() {
...
+    private fun postRequestSignature() {
+        val retrofitClient = RetrofitClient.getClient()
+        addDisposable(retrofitClient.postRequestSignature(establishDataValues)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .doOnTerminate {
+                initWidget()
+            }
+            .subscribe({
+                Toast.makeText(this, it.string(), Toast.LENGTH_SHORT).show()
```
[the remark this answers is not in the document]
Comment (andresguedes) [reply app/src/main/java/net/trustly/trustlysdkdemoandroid/ui/MainActivity.kt:61]: This Toast is just to show the requestSignature to the user, but I'll change it and put in a Log for better DevX =)

File: README.md [hunk 3]
```diff
@@ -40,6 +40,17 @@ object EstablishData {
 ```
 5. Setup a target device or emulator and "Run" the application
 
+### Request Signature
```
Comment (lukevance) [inline README.md:43]: 👍 Looks good!

**Review by lukevance — APPROVED** [review]
Awesome 😎

**Pull request merged**